tests/test_chebyMap.py

```diff
@@ -0,0 +1,493 @@
+
+        outPoints = chebyMap1.tranForward(inPoints)
+
+        referenceCheby = ReferenceCheby(referenceFunc, lbnd_f, ubnd_f)
+        desiredOutPoints = referenceCheby.transform(inPoints)
+
+        npt.assert_allclose(outPoints, desiredOutPoints)
+
+        with self.assertRaises(RuntimeError):
+            chebyMap1.polyTran(forward=False, acc=0.0001, maxacc=0.001, maxorder=6,
+                               lbnd = lbnd_f, ubnd=ubnd_f)
+
+    def test_chebyGetDomain(self):
+        """Test ChebyMap.getDomain's ability to estimate values
+
+        This occurs when there is only one map and you want the inverse
+        """
+        nout = 2
+        lbnd_f = [-2.0, -2.5]
+        ubnd_f = [1.5, 2.5]
+
+        # Coefficients for the following not-gently-varying polynomial:
+        # y1 = 2.0 T2(x1') T0(x2') - 2.0 T0(x1') T2(x2')
+        # y2 =  1.0 T3(x1') T0(x2') - 2.0 T0(x1') T3(x2')
+        coeff_f = np.array([
+            [2.0, 1, 2, 0],
+            [-2.0, 1, 0, 2],
+            [1.0, 2, 3, 0],
+            [-2.0, 2, 0, 3],
+        ])
+
+        chebyMap1 = astshim.ChebyMap(coeff_f, nout, lbnd_f, ubnd_f)
+
+        # compute inPoints as a grid of points that cover the input range
+        x1Edge = np.linspace(lbnd_f[0], ubnd_f[0], 1000)
+        x2Edge = np.linspace(lbnd_f[1], ubnd_f[1], 1000)
+        x1Grid, x2Grid = np.meshgrid(x1Edge, x2Edge)
+        inPoints = np.array([x1Grid.ravel(), x2Grid.ravel()]).transpose()
+        inPoints = inPoints.copy()  # make contiguous
+
+        outPoints = chebyMap1.tranForward(inPoints)
+        pred_lbnd = outPoints.min(0)
+        pred_ubnd = outPoints.max(0)
+
+        domain = chebyMap1.getDomain(forward=False)
+        npt.assert_allclose(domain.lbnd, pred_lbnd, atol=0.0001)
+        npt.assert_allclose(domain.ubnd, pred_ubnd, atol=0.0001)
+
+    def test_normalize(self):
+        """Test the local utility function `normalize`
+        """
+        lbnd = [-2.0, -2.5]
+        ubnd = [1.5, 2.5]
+
+        # points that cover the full domain
+        points = np.array([
+            [-2.0, -2.5],
+            [-0.5, 1.5],
+            [0.5, -0.5],
+            [1.5, 2.5],
+        ])
+
+        normPoints = normalize(points, lbnd, ubnd)
+        normByAxis = normPoints.transpose()
+        for normAxis in normByAxis:
+            self.assertAlmostEqual(normAxis.min(), -1)
+            self.assertAlmostEqual(normAxis.max(), 1)
+
+    def test_cheby(self):
+        """Test the local utility function `cheby`
+        """
+        points = np.linspace(-1, 1, 10, endpoint=True, dtype=float)
+        t0 = cheby(0, points)
+        npt.assert_allclose(t0, 1.0)
+        t1 = cheby(1, points)
+        t2 = cheby(2, points)
+        t3 = cheby(3, points)
+        t4 = cheby(4, points)
+        desired_t0 = np.ones(len(points))
+        desired_t1 = points
+        desired_t2 = 2 * points * t1 - t0
+        desired_t3 = 2 * points * t2 - t1
+        desired_t4 = 2 * points * t3 - t2
```

Comment:
Why not use [numpy](https://docs.scipy.org/doc/numpy/reference/generated/numpy.polynomial.chebyshev.chebval.html) as your oracle?